docs/getting-started/openshift/index.md

```diff
@@ -0,0 +1,195 @@
+
+For non-production scenarios, you can use the argocd-agentctl CLI's PKI tooling. Please be advised that this is not a production-grade setup, and that important pieces of the PKI, such as the CA's private key, will be stored unprotected on your control plane cluster.
+
+Create the PKI on the principal:
+
+```
+argocd-agentctl --principal-context <control plane context> pki init
+```
+
+Issue the certificate for the principal's gRPC service:
+The gRPC service is the service the agents will connect to. This service usually needs to be exposed to the outside world, or at least from your other clusters where the agents are running on. Typically, this means the service is exposed by a load balancer or a node port.
+
+```
+argocd-agentctl pki issue principal \
+    --principal-context <control plane context> \
+    --ip <ip addresses of principal> \
+    --dns <dns names of principal>
+```
+
+For the --ip and --dns values, you want to specify all addresses and DNS names that the principal will be reachable at for the agents, for example --ip 5.5.5.5 --dns my-principal.example.com. If there is a load balancer in front of the principal, make sure it will pass through traffic to the principal - otherwise, client certificates might not be used for authentication.
+
+Issue the certificate for the principal's resource proxy:
+The principal's resource proxy needs to be reachable by your Argo CD API server (argocd-server), which usually runs in the same cluster as the principal. Typically, this service should not be reachable outside the cluster and is exposed using a Kubernetes Service.
+
+```
+argocd-agentctl pki issue resource-proxy \
+    --principal-context <control plane context> \
+    --ip <ip addresses of resource proxy> \
+    --dns <dns names of principal>
+```
+
+Deploy principal on hub cluster using argocd-operator/gitops-operator using Argo CD CR given below
+
+```
+apiVersion: argoproj.io/v1beta1
+kind: ArgoCD
+metadata:
+  name: argocd
+spec:
+  controller:
+    enabled: false
+  argoCDAgent:
+    principal:
+      enabled: true
+      allowedNamespaces: 
+        - "*"
+      jwtAllowGenerate: true
+      auth: "mtls:CN=([^,]+)"
+      logLevel: "trace"
+      image: "<quay.io/user/argocd-agent:v1>"
+  sourceNamespaces:
+    - "agent-managed"
+    - "agent-autonomous"  					
+```
+
+The above CR should create all the necessary resource for Argo CD as well as argocd-agent principal in argocd namespace.
+
+Create argocd-redis secret, because principal looks for it to fetch redis authentication details.
+
+```
+oc create secret generic argocd-redis -n argocd --from-literal=auth="$(oc get secret argocd-redis-initial-password -n argocd -o jsonpath='{.data.admin\.password}' | base64 -d)"
+```
+
+## Setting up agent workload cluster 
+
+### Configure Argo-CD for Agent 
+ 
+Argo CD instance on Agent cluster
+
+Creating Argo CD instance for Workload/spoke cluster.
```

Comment:
I had to create an AppProject as well, during testing

```
apiVersion: argoproj.io/v1alpha1
kind: AppProject
metadata:
  name: default
  namespace: argocd
spec:
  clusterResourceWhitelist:
    - group: '*'
      kind: '*'
  destinations:
    - namespace: '*'
      server: '*'
  sourceRepos:
    - '*'
```

Reply:
I think agent does appProject sync, I can add this in the bottom of troubleshooting list

Reply:
I'll find the error for troubleshooting section